packages/astro/components/ViewTransitions.astro

```diff
@@ -220,23 +220,15 @@ const { fallback = 'animate' } = Astro.props as Props;
 		links.length && (await Promise.all(links));
 
 		if (fallback === 'animate') {
-			let isAnimating = false;
-			addEventListener('animationstart', () => (isAnimating = true), { once: true });
-
 			// Trigger the animations
 			document.documentElement.dataset.astroTransitionFallback = 'old';
+			const finished = Promise.all(document.getAnimations().map(a => a.finished));
```

Comment:
Nice! This is the API I was talking about. Support matrix looks good to me.

<img alt="getAnimations support matrix" src="https://github.com/withastro/astro/assets/7118177/e8c39cc1-fc45-4f17-8cab-e5ce907845a0">